### ISSUE TITLE
V5 whitelist/blacklist API

### ISSUE THREAD

**Issue by sterrenburg** [sterrenburg/flutterhole#76]
Since V5 fetching the whitelist with the old format throws a server error: 

- `{{ base_url  }}?list=white&auth={{ auth  }}`
- `{{ base_url  }}?list=black&auth={{ auth  }}`

These return a 500 status on my local Pi. Not sure if the API changed or my Pihole installation is funky.

The V5 dashboard seems to get its data `http://pi.hole/admin/scripts/pi-hole/php/groups.php` with some url encoded form data. 

If anyone manages to query their Pi properly, please let me know!

**Comment by p1r473** [comment]
Is this why the option to whitelist/blacklist seems to be gone in latest Flutterboard?
How can I test this for you?

**Comment by sterrenburg** [comment]
@p1r473 that is correct! Perhaps you can report the server response from your own Pi-hole.

Try running the following in your browser:

```
{{ base_url }}?list=white&auth={{ auth }}
```

This would look something like the following:

```
http://pi.hole/admin/api.php?list=white&auth=my_token
```

This _used_ to return a JSON object of the whitelists. On my V5 Pi-hole, the server crashes with a HTTP status code 500.

As a sidenote, the [official documentation](https://docs.pi-hole.net/database/gravity/example/#example-4-whitelisting) seems to indicate that the black/whitelisting commands are now only accessible via direct SQL. Not sure how this could be integrated into the app, but I hope there still is an API endpoint somewhere...

**Comment by p1r473** [comment]
Same here.
Have you opened a bug at the PiHole github?
https://192.168.1.186:445/admin/api.php?list=white&h=<APIKEY>
![image](https://user-images.githubusercontent.com/9235633/82336595-cc5e8480-99b8-11ea-9098-bd161e95eb28.png)

**Comment by p1r473** [comment]
Looks like pihole -w and pihole -b still work, The whitelist and blacklist commands.

**Comment by sterrenburg** [comment]
> Same here.
Have you opened a bug at the PiHole github?

Thank you for trying it out 👍. I will try and create an issue over at the Pi-hole team.

> Looks like pihole -w and pihole -b still work, The whitelist and blacklist commands.

These are CLI commands right? I've thought about using SSH in the app, but it sounds like a security nightmare.

**Comment by p1r473** [comment]
Yup those are CLI commands.
If you make a PiHole ticket I will back it up with my evidence.

**Comment by p1r473** [comment]
@sterrenburg can you link your PiHole GitHub issue so I can post my evidence there?

**Comment by sterrenburg** [comment]
> @sterrenburg can you link your PiHole GitHub issue so I can post my evidence there?

Just now I posted on the Pi-hole discourse, thank you for taking a look:

https://discourse.pi-hole.net/t/black-whitelisting-does-not-work-on-v5-api/33119

The new API is being worked on in a [Pi-hole pull](https://github.com/pi-hole/FTL/pull/659), I imagine that the current API interface will need to change on my end once it is released.

**Comment by elmicha** [comment]
[Here](https://discourse.pi-hole.net/t/black-whitelisting-does-not-work-on-v5-api/33119/4) a Pi-hole developer wrote that it is resolved in Pi-hole v5.1 and the current version seems to be 5.1.2, so I guess it's fixed now on the Pi-hole side.

**Comment by sterrenburg** [comment]
@elmicha thank you for bringing this to my attention! Assuming all goes well, I expect to add white-/blacklist support back in the upcoming release.
